core/src/test/java/org/infinispan/util/concurrent/BoundedConcurrentHashMapLIRSTest.java

```diff
@@ -0,0 +1,198 @@
+      byteArrayConditionalRemove(createComparingConcurrentMap(), true);
+   }
+
+   public void testByteArrayReplace() {
+      byteArrayReplace(createComparingConcurrentMap(), true);
+   }
+
+   public void testByteArrayPutIfAbsentFail() {
+      byteArrayPutIfAbsentFail(createComparingConcurrentMap(), true);
+   }
+
+   protected void byteArrayConditionalRemove(
+         ConcurrentMap<byte[], byte[]> map, boolean expectRemove) {
+      byte[] key = {1, 2, 3};
+      byte[] value = {4, 5, 6};
+      map.put(key, value);
+      byte[] removeKey = {1, 2, 3}; // on purpose, different instance required
+      byte[] removeValue = {4, 5, 6}; // on purpose, different instance required
+      if (expectRemove)
+         assertTrue(String.format(
+               "Expected key=%s to be removed", str(removeKey)),
+               map.remove(removeKey, removeValue));
+      else
+         assertNull(map.get(removeKey));
+   }
+
+   protected void byteArrayReplace(
+         ConcurrentMap<byte[], byte[]> map, boolean expectReplaced) {
+      byte[] key = {1, 2, 3};
+      byte[] value = {4, 5, 6};
+      map.put(key, value);
+      byte[] lookupKey = {1, 2, 3};
+      byte[] oldValue = {4, 5, 6}; // on purpose, different instance required
+      byte[] newValue = {7, 8, 9}; // on purpose, different instance required
+      boolean replaced = map.replace(lookupKey, oldValue, newValue);
+      if (expectReplaced)
+         assertTrue(String.format(
+               "Expected key=%s replace of oldValue=%s with newValue=%s to work",
+               str(lookupKey), str(oldValue), str(newValue)), replaced);
+      else
+         assertFalse(replaced);
+   }
+
+   protected void byteArrayPutIfAbsentFail(
+         ConcurrentMap<byte[], byte[]> map, boolean expectFail) {
+      byte[] key = {1, 2, 3};
+      byte[] value = {4, 5, 6};
+      map.put(key, value);
+      byte[] putKey = {1, 2, 3}; // on purpose, different instance required
+      byte[] newValue = {7, 8, 9};
+      byte[] previous = map.putIfAbsent(putKey, newValue);
+      if (expectFail)
+         assertTrue(String.format(
+               "Expected putIfAbsent for key=%s to fail", str(putKey)),
+               Arrays.equals(value, previous));
+      else
+         assertNull(previous);
+   }
+
+   protected ConcurrentMap<byte[], byte[]> createStandardConcurrentMap() {
+      return new ConcurrentHashMap<byte[], byte[]>();
+   }
+
+   protected ConcurrentMap<byte[], byte[]> createComparingConcurrentMap() {
+      return new BoundedConcurrentHashMap<byte[], byte[]>(
+            BoundedConcurrentHashMap.DEFAULT_MAXIMUM_CAPACITY,
+            BoundedConcurrentHashMap.DEFAULT_CONCURRENCY_LEVEL, Eviction.LIRS,
+            EQUIVALENCE, EQUIVALENCE);
+   }
+
+//
+//   TODO: Enable test when Comparing.compare() function for a byte[] has been created, and hence tree based hash bins can be used.
+//
+//   public void testByteArrayOperationsWithTreeHashBins() {
```

Comment:
This was a remnant from the previous BoundedConcurrentHashMapTest class.  I will clean these up before review.